draft-ietf-jose-hpke-encrypt.md

```diff
@@ -253,15 +254,53 @@ Otherwise, the JWE Protected Header (and JWE Shared Unprotected Header) MUST NOT
 - JOSE Header parameter "psk_id" MAY be present.
 - JOSE Header parameter "ek" MUST be present and contain the base64url-encoded HPKE encapsulated key.
 - Recipient JWE Encrypted Key MUST be the ciphertext from HPKE Encryption.
-- The HPKE info parameter defaults to the empty string; mutually known private information MAY be used instead.
-- The HPKE AAD parameter MUST be set to the empty string.
+- The HPKE info parameter contains the encoding of the Recipient_structure, which is described in {{recipient_structure}}.
+- The HPKE AAD parameter defaults to the empty string; externally provided information MAY be used instead.
 - THE HPKE plaintext MUST be set to the CEK.
 
 The processing of "enc", "iv", "tag", "aad", and "ciphertext" is as already defined in {{RFC7516}}.
 Implementations process these parameters as defined in {{RFC7516}};
 no additional processing requirements are introduced by HPKE-based key encryption.
 
-## JSON Example
+## Recipient_structure {#recipient_structure}
+
+The `Recipient_structure` is a JSON object with the following members:
+
+- context (string): This member MUST include the constant string value "JOSE HPKE Recipient".
+
+- next_layer_alg (string): Identifies the algorithm with which the HPKE-encrypted key MUST be used. Its value MUST match the "enc" (encryption algorithm) header parameter in the next lower JOSE layer.
```

Comment:
we should justify the reason, it will address the threat discussed in https://datatracker.ietf.org/doc/rfc9709/ (which is applicable to both AES-CCM and AES-GCM, JWE uses AES-GCM).

Reply:
This field is there because of the alignment with COSE HPKE. If nobody ever uses non-AEAD ciphers with JOSE then I do not think this field is strictly necessary. So, it is a bit of a defense in depth approach here.

Reply:
Adding: I am not aware of an attack that works with just an AEAD cipher.

Reply:
I don’t think non-AEAD ciphers will be registered in the future. Perhaps add a note saying: “This field is included only for alignment with COSE HPKE.”

Reply:
Done. I have added this sentence.